bundles/mapping/mapmodule/plugin/vectortilelayer/VectorTileLayerPlugin.js

```diff
@@ -89,16 +99,74 @@ class VectorTileLayerPlugin extends AbstractMapLayerPlugin {
         if (options.tileGrid) {
             sourceOpts.tileGrid = new TileGrid(options.tileGrid);
         }
+        // Properties id, type and hover are being used in VectorFeatureService.
         const vectorTileLayer = new olLayerVectorTile({
             opacity: layer.getOpacity() / 100,
             renderMode: 'hybrid',
             source: new olSourceVectorTile(sourceOpts),
-            style: this._getLayerCurrentStyleFunction(layer)
+            style: this._getLayerCurrentStyleFunction(layer),
+            [LAYER_ID]: layer.getId(),
+            [LAYER_TYPE]: layertype,
+            [LAYER_HOVER]: layer.getHoverOptions()
```

Comment:
The ol/layer/VectorTile constructor options has a specific type signature:
http://openlayers.org/en/v5.2.0/apidoc/module-ol_layer_VectorTile-VectorTileLayer.html#VectorTileLayer

We shouldn't use the OL options-object as storing place for "extra" layer state. The options is meant for giving options for OL internal use. There is no guarantee these "extra" options will be stored in future versions of OL. In any case the options have a specific type and we should stick to that.

Reply:
The documentation states: "Note that any property set in the options is set as a module:ol/Object~BaseObject property on the layer object; for example, setting title: 'My Title' in the options means that title is observable, and has get/set accessors."

Reply:
The above excerpt just means that each valid options key has corresponding setter/getter on the layer instance. For example key `style` -> `getStyle()` on the layer. It's not meant as an endorsement for adding keys not listed in the type signature.